cmd/emailreport/mail.go

```diff
@@ -0,0 +1,85 @@
+/// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"log"
+	"net/smtp"
+	"strings"
+)
+
+// headerRFC822 provides a header as a
+// string according the RFC822 Standard for ARPA
+// Internet Text Messages.
+func HeaderRFC822(conf Configuration) string {
+	var header string
+
+	if conf.Mail.From == "" {
+		log.Fatal("Sender no specified")
+	}
+
+	if conf.Mail.Subject == "" {
+		log.Fatal("Message subject no specified")
+	}
+
+	if len(conf.Mail.To) == 0 {
+		log.Fatal("Receipts no specified")
+	}
+
+	header += "From: " + conf.Mail.From + "\n"
+	header += "To: " + strings.Join(conf.Mail.To, ",") + "\n"
+
+	if len(conf.Mail.Cc) > 0 {
+		header += "cc: " + strings.Join(conf.Mail.Cc, ",") + "\n"
+	}
+
+	header += "Subject: " + conf.Mail.Subject + "\n\n"
+
+	return header
+}
+
+// SendByEmail allows to send the result output of
+// checkmetrics execution to a whitelist of emails.
```

Comment:
I think we may have missed the intent/meaning of the whitelisting here, and how we are going to implement it. It's doesn't have to be a blocker, and I will open a new Issue to add the 'whitelist checking' as a next step to this tool.
The notion of the whitelist is that we should only send emails to a defined list of 'allowed' addresses. And we have chosen that rather than have an actual list in a file that defines legal destination email adresses, we will use the github 'organisation' list as the list to check (that is, we will only send out emails to PRs submitted by members of the organisation). Let me put that all in a new Issue.